src/diffusers/models/transformers/lumina_nextdit2d.py

```diff
@@ -0,0 +1,608 @@
+            bias=True,
+        )
+
+        self.pad_token = nn.Parameter(torch.empty(hidden_size))
+
+        self.time_caption_embed = LuminaCombinedTimestepCaptionEmbedding(
+            hidden_size=min(hidden_size, 1024), encoder_hidden_size=encoder_hidden_size
+        )
+
+        self.layers = nn.ModuleList(
+            [
+                LuminaNextDiTBlock(
+                    layer_id,
+                    hidden_size,
+                    num_attention_heads,
+                    num_kv_heads,
+                    multiple_of,
+                    ffn_dim_multiplier,
+                    norm_eps,
+                    qk_norm,
+                    encoder_hidden_size,
+                )
+                for layer_id in range(num_layers)
+            ]
+        )
+        self.final_layer = LuminaFinalLayer(hidden_size, patch_size, self.out_channels)
+
+        assert (hidden_size // num_attention_heads) % 4 == 0, "2d rope needs head dim to be divisible by 4"
+        self.freqs_cis = self.precompute_freqs_cis(
+            hidden_size // num_attention_heads,
+            384,
+            scaling_factor=scaling_factor,
+        )
+
+    def unpatchify(self, x: torch.Tensor, img_size: List[Tuple[int, int]], return_tensor=False) -> List[torch.Tensor]:
+        """
+        Reconstructs the original images from the patchified tensor.
+
+        Args:
+            x (torch.Tensor): The patchified tensor of shape (N, T, patch_size**2 * C).
+            img_size (List[Tuple[int, int]]): The list of image sizes for each image in the batch.
+            return_tensor (bool, optional): Whether to return the reconstructed images as a tensor.
+                If False, the reconstructed images will be returned as a list of tensors. Defaults to False.
+
+        Returns:
+            List[torch.Tensor] or torch.Tensor: The reconstructed images.
+                If return_tensor is True, the reconstructed images will be returned as a tensor of shape (N, C, H, W).
+                If return_tensor is False, the reconstructed images will be returned as a list of tensors, where each
+                tensor has shape (H, W, C).
+        """
+        pH = pW = self.patch_size
+        if return_tensor:
+            H, W = img_size[0]
+            B = x.size(0)
+            L = (H // pH) * (W // pW)
+            x = x[:, :L].view(B, H // pH, W // pW, pH, pW, self.out_channels)
+            x = x.permute(0, 5, 1, 3, 2, 4).flatten(4, 5).flatten(2, 3)
+            return x
+        else:
+            imgs = []
+            for i in range(x.size(0)):
+                H, W = img_size[i]
+                L = (H // pH) * (W // pW)
+                imgs.append(
+                    x[i][:L]
+                    .view(H // pH, W // pW, pH, pW, self.out_channels)
+                    .permute(4, 0, 2, 1, 3)
+                    .flatten(3, 4)
+                    .flatten(1, 2)
+                )
+        return imgs
+
+    def patchify_and_embed(
```

Comment:
I think we can make a `LuminaPatchEmbed`, no? 
I think it does similar things as `PathEmbed`, let me know if it's not https://github.com/huggingface/diffusers/blob/8690e8b9d6e68b3aec28f7f84374dd8256e79f18/src/diffusers/models/embeddings.py#L125

Reply:
I don't think we can use PatchEmbed because our patches obtain their embeddings using a linear layer.